2주차/20920.py

```diff
@@ -0,0 +1,12 @@
+N,M = map(int,input().split())
+word =[input() for _ in range(N)]
+for w in word :
+    if len(w) < M :
+        word.remove(w)
```

Comment:
```suggestion
word =[]
for _ in range(N):
    w = input()
    if len(w) >= M:
        word.append(w)
```
list의 remove 연산은 리스트의 첫 번째 원소부터 하나씩 확인하며 삭제할 원소를 찾고, 해당 원소를 찾아 삭제하고 나면 그 뒤에 있는 원소들을 전부 한 칸 씩 앞으로 당겨오는 식으로 동작하기 때문에 시간복잡도가 O(n)이에요.
따라서 작성해준 for문은 시간복잡도가 O(n^2)으로 비효율적이니 이럴때는 수정한 코드처럼 리스트 컴프리헨션을 사용하지 않고 O(n)의 시간으로 입력받아야해요